File: GoRouterExample/app/src/main/java/cn/gorouter/gorouterexample/MainActivity.java
package cn.gorouter.gorouterexample;

import androidx.annotation.Nullable;

import android.content.Intent;
import android.os.Bundle;

import cn.gorouter.annotation.Route;
import cn.gorouter.api.launcher.GoRouter;
import cn.gorouter.api.logger.GoLogger;
import cn.gorouter.common.library.BaseActivity;

/**
 * @author logcat
 */
@Route(url = "/main/MainActivity")
public class MainActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        GoRouter.getInstance()
                .build("/another/AnotherActivity")
                .go();
    }


    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        switch (requestCode){
            case 1001:
                GoLogger.info("Back from another Activity!!!");
                break;
            default:
                break;
        }
    }
}

File: gorouter-kotlin-demo-main/src/main/java/cn/rubintry/gorouter/kotlin/MainActivity.kt
package cn.rubintry.gorouter.kotlin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import cn.gorouter.api.kotlin.core.GoRouter
import cn.rubintry.annotation.kotlin.Route
import cn.rubintry.gorouter_kotlin_demo_lib_common.service.ICommonService


@Route(path = "/Demo/MainActivity")
class MainActivity : AppCompatActivity() {
    private val TAG = this.javaClass.simpleName



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val instance = GoRouter.getInstance().build("/Demo/MFragment").go()
        instance?.let {
            print("Fragment instance ${it.javaClass.name}")
        }


//        val instance = GoRouter.getInstance().build("/Service/TargetService").go(ICommonService::class.java)
//        instance.common(this)
    }


    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if(requestCode == 10){
            Log.d(TAG, "onActivityResult: ")
        }
    }

    fun goToSecond(view: View) {
        GoRouter.getInstance()
                .build("/test/SecondActivity")
                .withString("key" , "value1")
                .go()
    }
}